corgidrp/spec.py

```diff
@@ -0,0 +1,629 @@
+          
+        cfamname.append (frames.ext_hdr['CFAMNAME'])
+        slits.append (fsamname)
+    if len(np.unique(slits)) != 1:
+        raise AttributeError("currently we only have template files for no slit or R1C2, not for "+ slits)
+    if len(np.unique(cfamname)) == 1:
+        band = cfamname[0]
+        if not band.startswith ("3"):
+            raise AttributeError("currently we only have template files for the filter band 3, not for "+ band)
+        slit = slits[0]
+        if slit == "R1C2":
+            filenames = sorted(glob.glob(os.path.join(template_dir,"spec_unocc_r1c2slit_offset_prism3_3d_*.fits")))
+        elif slit == "OPEN":
+            filenames = sorted(glob.glob(os.path.join(template_dir,"spec_unocc_noslit_offset_prism3_3d_*.fits")))
+        else:
+            raise AttributeError("we do not (yet) have template files for slit " + slit)
+    else:
+        #filtersweep
+        filenames = sorted(glob.glob(os.path.join(template_dir, "spec_unocc_noslit_offset_prism3_filtersweep_*.fits")))
+        filtersweep = True
+    return Dataset(filenames), filtersweep
+            
+
+def compute_psf_centroid(dataset, template_dataset = None, initial_cent = None, filtersweep = False, halfwidth=10, halfheight=10, verbose = False):
+    """
+    Compute PSF centroids for a grid of PSFs and return them as a calibration object.
+
+    Args:
+        dataset (Dataset): Dataset containing 2D PSF images. Each image must include pri_hdr and ext_hdr.
+        template_dataset (dataset): dataset of the template PSF, if None, a simulated PSF from the data/spectroscopy/template path is taken
+        initial_cent (dict): Dictionary with initial guesses for PSF centroids.
+                             Must include keys 'xcent' and 'ycent', each mapping to an array of shape (N,).
+        filtersweep (bool): If True, it uses a filter sweep/scan dataset.
+        halfwidth (int): Half-width of the PSF fitting box.
+        halfheight (int): Half-height of the PSF fitting box.
+        verbose (bool): If True, prints fitted centroid values for each frame.
+    
+    Returns:
+        SpectroscopyCentroidPSF: Calibration object with fitted (x, y) centroids.
+    """
+    if not isinstance(dataset, Dataset):
+        raise TypeError("Input must be a corgidrp.data.Dataset object.")
+    
+    if initial_cent is None:
+        xcent, ycent = None, None
+    else:
+        xcent = np.asarray(initial_cent.get("xcent"))
+        ycent = np.asarray(initial_cent.get("ycent"))
+        if xcent is None or ycent is None:
+            raise ValueError("initial_cent dictionary must contain 'xcent' and 'ycent' arrays.")
+        if len(dataset) != len(xcent) or len(dataset) != len(ycent):
+            raise ValueError("Mismatch between dataset length and centroid guess arrays.")
+    
+    if template_dataset is None:
+        template_dataset, filtersweep = get_template_dataset(dataset)
```

Comment:
It seems like this line overrides the filtersweep argument passed in. Perhaps the filtersweep docstring comment could emphasize that it's only used if template_dataset is not None?